### PR TITLE
test(ocap-kernel): Fix flaky network tests by mocking classes instead of libp2p

### PR DESCRIPTION
**Pull request by sirtimid**
Simplified network tests to mock the extracted classes rather than libp2p, eliminating flakiness. Tests now focus on `network.ts` functionality and cover basic messaging, connections, and reconnection logic. Also improved coverage of `kernel-utils` package.

---

> [!NOTE]
> Adds comprehensive tests for `kernel-utils` (hex/retry) and rewrites `ocap-kernel` network tests to mock internal classes, simplifies `abortableDelay`, and adjusts Vitest coverage thresholds.
> 
> - **Kernel Utils**:
>   - **Tests**: Add `packages/kernel-utils/src/hex.test.ts`; greatly expand `retry.test.ts` to cover defaults, jitter/backoff, abort signaling, and edge cases.
>   - **Abortable Delay**: Simplify `abortableDelay` in `misc.ts` (remove post-listener aborted recheck); align tests by removing race-condition specific cases from `misc.test.ts`.
>   - **Coverage**: Add `/* v8 ignore */` guards in `retry.ts` for unreachable paths.
> - **OCAP Kernel**:
>   - **Network Tests**: Replace libp2p-heavy mocks with class-based mocks (`MessageQueue`, `ReconnectionManager`, `ConnectionFactory`, etc.) in `remotes/network.test.ts`; add scenarios for basic messaging, inbound handling, reconnection, queue flushing, wake events, stop/cleanup, race conditions, and error handling.
> - **Tooling**:
>   - **Vitest thresholds**: Update coverage thresholds (e.g., `packages/kernel-utils/**` to 100%; minor adjustments for `packages/kernel-platforms/**` and `packages/ocap-kernel/**`).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 4f8650a6b0dff41db8701545f01211543cc611d6. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/ocap-kernel/src/remotes/network.test.ts [hunk 1]
```diff
@@ -1,2236 +1,1135 @@
...
+  vi,
+  beforeEach,
+  afterEach,
+  beforeAll,
+} from 'vitest';
+
+import type { QueuedMessage } from './MessageQueue.ts';
+
+// Import the module we're testing - must be after mocks are set up
+let initNetwork: typeof import('./network.ts').initNetwork;
```
[the remark this answers is not in the document]
Comment (rekmarks) [reply packages/ocap-kernel/src/remotes/network.test.ts]: Ah, or is the issue that you're declaring mocks in this file that you're then using inside of the `vi.mock` calls? I guess that's fine then.

Elsewhere, we've used `vi.hoisted()` to obtain a reference to our mocks and use them inside of `vi.mock()`.
